core/tests/webdriverio_utils/BlogDashboardPage.js

```diff
@@ -243,33 +226,40 @@ var BlogDashboardPage = function() {
 
   this.expectNumberOfPublishedBlogPostsToBe = async function(number) {
     await this.waitForPublishedBlogPostsToLoad();
-    expect(await publishedBlogPostTiles.count()).toBe(number);
+    var publishedBlogPostTiles = await publishedBlogPostTilesSelector();
+    expect(publishedBlogPostTiles.length).toBe(number);
   };
 
   this.expectNumberOfBlogPostsRowsToBe = async function(number) {
     await this.waitForDraftBlogPostsToLoad();
-    expect(await blogPostListItems.count()).toBe(number);
+    var blogPostListItems = await blogPostListItemsSelector();
+    expect(blogPostListItems.length).toBe(number);
   };
 
   this.getBlogPostTileEditOption = async function(title) {
     await waitFor.visibilityOf(
-      blogPostTiles.first(), 'Blog Post tiles take too long to be visible.');
-    for (i = 0; i < await blogPostTiles.count(); i++) {
-      var blogPostTile = blogPostTiles.get(i);
-      var blogPostTitleContainer = blogPostTile.element(
-        blogPostTileTitle);
+      blogPostTileElement, 'Blog Post tiles take too long to be visible.');
+    var blogPostTiles = await blogPostTilesSelector();
+    for (i = 0; i < blogPostTiles.length; i++) {
+      var blogPostTile = blogPostTiles[i];
+      var blogPostTitleContainer = await blogPostTile.$(
+        '.e2e-test-blog-post-title');
+      await waitFor.elementToBeClickable(
+        blogPostTitleContainer, 'Blog Post title is not interactable');
```

Comment:
In webdriverio even if the element is visible but its not interactable, we get null as a result of using getText() on it ([link](https://webdriver.io/docs/api/element/getText)). So I am using elementToBeClickable instead of visibiltyOF to make sure its interactable before calling getText() on it. 
P.S As we discussed in the previous meeting there is no function that only checks for interactability of element that I know so I am using elementToBeClickable as a best possible solution.

Reply:
So, should it not be added inside the hood of action.getText() only?

Reply:
Doing so fails other test as we actually clicks on element again and again to check whether its clicakable or not and this creates an issue for other tests.

Reply:
At every where else the element becomes interactable while we wait for it to be visible but here even if we wait for the element to be visible its still remains uninteractable, hence we explicitly needs to wait for it to become interactable. Also elementToBeClickable is not the right function that can be used to wait for an element to be interactable (there is no function that actually checks interactability only), so if we will use this in elementToBeClickable action.js we gets the following error: Basically it clicks again and again to check whether its clicakble or not, which creates this error in moveToState function where we are using getText on stateName elements.

https://user-images.githubusercontent.com/86483059/181156924-d382861b-c60f-450f-ad10-1e2f9096a004.mp4